src/persist-types/src/columnar.rs

```diff
@@ -130,6 +131,15 @@ pub trait Schema2<T>: Debug + Send + Sync {
 
     /// Returns a type that can encode values of `T`.
     fn encoder(&self) -> Result<Self::Encoder, anyhow::Error>;
+
+    /// Returns the data type of arrays generated by this schema.
+    fn data_type(&self) -> anyhow::Result<DataType> {
```

Comment:
This does the opposite of what the internal comment says :). Let's pull it into a free function, something like `fn data_type<C: Codec>(schema: &C::Schema) -> anyhow::Result<DataType>`